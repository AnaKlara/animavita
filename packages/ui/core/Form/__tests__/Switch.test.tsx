import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent} from '@testing-library/react-native';

import {Mount, ThemeContext} from '../../../tests/helpers';
import Switch from '../Switch';

describe('Switch', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('calls onValueChange', () => {
    const cb = jest.fn();
    const {getByLabelText} = Mount(<Switch onValueChange={cb} accessibilityLabel="switch" />);

    fireEvent(getByLabelText('switch'), 'valueChange');

    expect(cb).toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const tree = renderer.create(ThemeContext(<Switch />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
