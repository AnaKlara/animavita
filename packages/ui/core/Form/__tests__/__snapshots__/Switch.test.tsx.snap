// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders correctly 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#2ED68F"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Object {},
      ],
    ]
  }
  thumbTintColor="#2ED68F"
  tintColor="#BABABA"
  value={false}
/>
`;
